tests/filters.rs

```diff
@@ -33,15 +41,14 @@ impl Log for Logger {
 
 fn main() {
     let mut a = None;
-    set_logger(|max| {
-                   let me = Arc::new(State {
-                                         last_log: Mutex::new(None),
-                                         filter: max,
-                                     });
-                   a = Some(me.clone());
-                   Box::new(Logger(me))
-               })
-            .unwrap();
+    try_set_logger(|max| {
```

Comment:
If the test only tests the `try_` version, you can probably remove the `set_logger` def at line 18 and the `set_logger` import in line 9